src/core/a-entity.js

```diff
@@ -435,14 +435,28 @@ var proto = Object.create(ANode.prototype, {
       if (isDefault || isMixedIn) { return; }
 
       component = this.components[name];
+
+      if (!component) { return; }
+
+      // Wait for component to initialize.
+      if (!component.initialized) {
```

Comment:
Maybe it's best to try to make sure that component gets cleaned up. Might cause leaks or unexpected behavior if we let it be orphaned in the case entities are being attached/detached before the component initializes?